README.md

```diff
@@ -1,42 +1,132 @@
-2. Run `dockercloud/haproxy` outside Docker Cloud: When you link containers to `dockercloud/haproxy`, the link alias matters. Any link alias sharing the same prefix and followed by "-/_" with an integer is considered from the same service. For example: `web-1` and `web-2` belong to service `web`, `app_1` and `app_2` are from service `app`, but `app1` and `web2` are from different services.
+That's it - the haproxy container will start querying Docker Cloud's API for an updated list of containers in the service and reconfigure itself automatically, including:
 
-Configuration
--------------
+* start/stop/terminate containers in the linked application services
+* start/stop/terminate/scale up/scale down/redeploy the linked application services
+* add new links to HAProxy
+* remove old links from HAProxy
 
-###Global and default settings of HAProxy###
+##### example of stackfile in Docker Cloud:
+
+	web:
+	  image: 'dockercloud/hello-world:latest'
+	  target_num_containers: 2
+	lb:
+	  image: 'dockercloud/haproxy:latest'
+	  links:
+	    - web
+	  ports:
+	    - '80:80'
+	  roles:
+	    - global
+
+### Running with docker legacy links
+
+Legacy link refers to the link created before docker 1.10, and the link created in default bridge network in docker 1.10 or after.
+
+##### example of legacy links using docker cli
+
+    docker run -d --name web1 dockercloud/hello-world
+    docker run -d --name web2 dockercloud/hello-world
+    docker run -d -p 80:80 --link web1:web1 --link web2:web2 dockercloud/haproxy
+
+#### example of docker-compose.yml v1 format:
+
+	web1:
+	  image: 'dockercloud/hello-world:latest'
+	web2:
+	  image: 'dockercloud/hello-world:latest'
+	lb:
+	  image: 'dockercloud/haproxy:latest'
+	  links:
+	    - web1
+	    - web2
+	  ports:
+	    - '80:80'
+
+**Note**: The link alias matters. Any link alias sharing the same prefix and followed by "-/_" with an integer is considered to be from the same service. For example: `web-1` and `web-2` belong to service `web`, `app_1` and `app_2` are from service `app`, but `app1` and `web2` are from different services.
+
+
+### Running with docker-compose v2(new links)
+
+Docker Compose 1.6 starts to support a new format of the compose file. In the new version(v2), the old link that injects environment variables is deprecated. Similar to using legacy links, here list some differences that you need to notice:
+
+- This image must be run using Docker Compose, as it relies on the Docker Compose labels for configuration
+- The container need access to the docker socket, you must mount the correct files and set the related environment to make it work
+- Although link is not needed from docker 1.10 for service discovery, you still need to set the link, so that `dockercloud/haproxy knows witch service it needs to balance. You don't need to set the linked alias, as it not used.
```

Comment:
A link is required in order to ensure that `dockercloud/haproxy` is aware of which service it needs to balance, although links are not needed for service discovery since docker 1.10. Linked aliases are not required.